dandi/files/bids.py

```diff
@@ -154,7 +171,7 @@ def bids_path(self) -> str:
         """
         ``/``-separated path to the asset from the root of the BIDS dataset
         """
-        return self.filepath.relative_to(self.bids_root).as_posix()
+        return self.filepath.resolve().relative_to(self.bids_root).as_posix()
```

Comment:
the `relative_to` check fail without it:
```
Error: '/home/chymera/.local/share/datalad/000108/dataset_description.json' is not in the subpath of '000108' OR one path is relative and the other is absolute.
```

Reply:
The reason that error is happening is because of [bidsschematools using `os.path.abspath`](https://github.com/bids-standard/bids-specification/blob/d0d04a1deb7adcd2552f19ff9539cd11046a02a2/tools/schemacode/bidsschematools/validator.py#L62).  I think it would make more sense for bidsschematools to leave the supplied paths as-is so that the caller of the library can easily match the output paths to the input paths.  (Also, while we're on the subject, I also believe it's wrong for the library to call `expanduser()`; resolving tildes should be the caller's job.)

Reply:
@jwodder  the problem with that is that BST should be fully usable in and of itself. It could include an additional wrapper which would be called for standalone usage to make paths absolute, but (1) that would make it more nested, and (2) I foresee that a wrapper will attract more ancillary features and become less comparable to DANDI which would be skipping the wrapper.

I guess I could come up with a non-wrapper solution where I store the original path, and de-absolutize everything on the back end depending on how the input was provided... though I get the sense that that sort of thing would just be more confusing, with people getting relative paths in the log files which can't be copy-pasted if they've moved to a new dir. I guess my logic was that as long as we're dealing with paths that actually exist, there's really no reason for them to ever be relative.

Is there a specific DANDI use case I would be compromising if I use `.absolute()` as in [242be00](https://github.com/dandi/dandi-cli/pull/1154/commits/242be00ebcfd722947279dbc7b972e08bf27b6a1)?

Reply:
@TheChymera

> Ultimately the rationale is that as long as we're dealing with paths that actually exist, there's really no reason for them to ever be relative.

I disagree.  If the user gives you relative paths, you should operate on those as-is unless you actually need the full paths.  Do you?

Reply:
> unless you actually need the full paths. Do you?

we do, unless we are provided with a potentially fictional list of paths, which is what the `accept_dummy_paths` argument does. Otherwise we need to walk through them so tilde will in any case break that functionality.

> leave the paths in Dandi alone as much as possible and only absolutize [here](https://github.com/dandi/dandi-cli/pull/1154/files#diff-70a4bb759311189af6f4468d78254d863319fc4660ebe437060b6634c5a7b724L82).

ok, thank you :)